app/models/course/reference_time.rb

```diff
@@ -7,16 +7,33 @@ class Course::ReferenceTime < ApplicationRecord
   validates :reference_timeline, presence: true, uniqueness: { scope: :lesson_plan_item }
   validates :lesson_plan_item, presence: true
 
-  validate :validate_start_at_cannot_be_after_end_at
+  validate :start_at_cannot_be_after_end_at
+  validate :lesson_plan_item_in_same_course
+
+  before_destroy :prevent_destroy_if_in_default_timeline, prepend: true
```

Comment:
I guess in retrospect, the `prepend: true` doesn't really do much in this model because it doesn't have any children with `dependent: :destroy`. But in the off chance that we start adding such children to this model, we can always be sure that the `before_` in `before_destroy` is always executed _before_ destroy.

It could save us from a headache should we [encounter something like this](https://github.com/rails/rails/issues/670).